Add TensorFlow Lite

margaretmz opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**[https://github.com/margaretmz/awesome-tensorflow-lite](https://github.com/margaretmz/awesome-tensorflow-lite)**

**About**
This repo contains a list of TensorFlow Lite models with sample apps, tutorials, and resources to showcase what the community has built and help you learn on-device ML.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

margaretmz commented:
unicorn

margaretmz commented:
I reviewed the following PRs: [1864](https://github.com/sindresorhus/awesome/pull/1864), [1863](https://github.com/sindresorhus/awesome/pull/1863), & [1797](https://github.com/sindresorhus/awesome/pull/1797).

margaretmz commented:
My list has 2 awesome-lint errors (which I don't think need fixing),  reported in issues [120](https://github.com/sindresorhus/awesome-lint/issues/120) & [121](https://github.com/sindresorhus/awesome-lint/issues/121).

margaretmz commented:
Fixed all errors reported by awesome-lint.

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

margaretmz commented:
> It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

Thanks for your review feedback. I updated the PR title to match "Add Name of List" and also updated the PR description. 
I went through the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) quite a few more times and didn't catch other issues. Could give me some examples of my mistakes please?

margaretmz commented:
Updated the PR: replacing entry description with what is TensorFlow Lite instead of what the awesome list is about.

axsaucedo commented:
One suggestion, I think you have to append to end of list (or in this case sub-list) when adding a new entry

margaretmz commented:
> One suggestion, I think you have to append to end of list (or in this case sub-list) when adding a new entry

Thanks @axsaucedo so much! I really appreciate your review and specific feedback.

I understand the guideline as "Your entry should be added at the bottom of the appropriate category" although I think it makes much more sense to have TensorFlow Lite below TensorFlow.js instead of below Education because they are all TensorFlow products:
* Deep Learning 
  * TensorFlow
  * TensorFlow.js
  * TensorFlow Lite
  * Papers 
  * Education

@sindresorhus what is your opinion on this? I'm OK to stick to the guideline strictly if that's the preference although in my opinion it looks really strange to list "TensorFlow Lite" below "Education". 

On a slightly related topic, which is outside of the scope of this PR but related to the topic of placement of new entries, I think "Deep Learning" should be a sub-bullet point of "Machine Learning" and "Deep Vision" should be under "Deep Learning".

margaretmz commented:
To help my reviewers, here is a summary of why I think I met the guidelines:

My awesome list: [https://github.com/margaretmz/awesome-tflite](https://github.com/margaretmz/awesome-tflite)

> You have to review at least 2 other open pull requests. 

I have reviewed the following PRs: [1864](https://github.com/sindresorhus/awesome/pull/1864), [1863](https://github.com/sindresorhus/awesome/pull/1863), [1797](https://github.com/sindresorhus/awesome/pull/1797) & [1859](https://github.com/sindresorhus/awesome/pull/1859)

> You have read and understood the instructions for creating a list.

Yes

> This pull request has a title in the format `Add Name of List`.
Name of List is TFLite so my PR is titled as 'Add TFLite'

> Your entry here should include a short description about the project/theme of the list.

My entry describes what is TFLite: "TensorFlow Lite - A set of tools to help convert and optimize TensorFlow models for on-device training and inference."

> Your entry should be added at the bottom of the appropriate category.

I added my entry below tensorflow.js which I think is a logic place for it.

> The suggested Awesome list complies with the below requirements.
> Has been around for at least 30 days.

My awesome list has been around since 2/28/2020

> Don't open a Draft / WIP pull request while you work on the guidelines. 

I opened the PR when I believed that I met the guidelines although I did make updates afterwards.

> Run `awesome-lint` on your list and fix the reported issues.

0 issues reported by `awesome-lint` as of today.

> The default branch should be named `main`, not master.

Yes default branch is `main`

> Includes a succinct description of the project/theme at the top of the readme. 

Yes it's included.

> Put in considerable effort into your list

Yes I have put into considerable effort into my list (take a look at my [commits](https://github.com/margaretmz/awesome-tflite/commits/main)) and also this PR.

> The repo name of your list should be in lowercase slug format

Yes `awesome-tflite`.

> The heading title of your list should be in title case format: # Awesome Name of List.

Yes `# Awesome TFLite`.

> Non-generated Markdown file in a GitHub repo.

Met requirement.

> The repo should have `awesome-list` & `awesome` as GitHub topics. 

Yes both `awesome-list` and `awesome` are included as GitHub topics

> Not a duplicate. 

My awesome-tflite list is not a duplicate of any other awesome lists.

> Only has awesome items. 

Yes.

> Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs.

Yes I keep the list up to date and awesome-lint also helped detecting broken links.

> Includes a project logo/illustration whenever possible.
 
I designed an illustration summarize what my awesome list is about. It's not a project logo so I didn't link to any project website or any relevant website.

> Entries have a description

Yes most entries have a description, except for a few links to official documentation which is self explanatory. 

> Includes the Awesome badge.

Yes the Awesome badge is included, placed on the right side of the readme heading and linked back to this list.

> Has a Table of Contents section 
 
Yes my list has a TOC named `Contents` as the first section in the list with only one level of nested lists.

> Has an appropriate license.

Yes my list has the CC0 license and to verify that I have read all the guidelines, I also comment on your PR with just the word unicorn.

> Has contribution guidelines.

Yes

> Has consistent formatting and proper spelling/grammar.

Yes.

> The link and description are separated by a dash.
> The description starts with an uppercase character and ends with a period.

Yes and verified with `awesome-lint`.

> Consistent and correct naming. 
> Doesn't use hard-wrapping.
> Doesn't include a Travis badge.
> Doesn't include an Inspired by awesome-foo or Inspired by the Awesome project kinda link at the top of the readme. The Awesome badge is enough.

Yes my list met these requirements.

guifelix commented:
The only suggestion I think I could find is that your repo have a banner (with all sort of info about it) instead of a logo on the top. Everything else seems ok

margaretmz commented:
> The only suggestion I think I could find is that your repo have a banner (with all sort of info about it) instead of a logo on the top. Everything else seems ok

Thanks @guifelix for your feedback! I appreciate you reviewing my PR.

I interpret the requirement of `Includes a project logo/illustration whenever possible.` as:
- "whenever possible" indicate it's an *optional* requirement
- it can be a project logo or illustration. And  in this case I created an illustration summarizing what "Awesome TFLite" is about and where/how TFLite can be used.

margaretmz commented:
Does anyone know what I need to do in order to get this PR approved and merged?

sindresorhus commented:
> The image should be high-DPI. Set it to maximum half the width of the original image.

I also think the header image looks a bit noisy. I would simplify it.

---

`TensorFlow Lite` in `TensorFlow Lite (TFLite) is a set of tools that help convert and optimize TensorFlow` should be linkified to its website.

---

> Consistent and correct naming. For example, Node.js, not NodeJS or node.js.

The official name is `TensorFlow Lite`, which should be used everywhere.

---

The formatting in `ML Kit examples` is inconsistent and could also be improved.

---

Date format should be `2020-04-20`, not `4/20/2020`, to remove ambiguity.

---

> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.

sindresorhus commented on readme.md:
```diff
@@ -336,6 +336,7 @@
 - [Deep Learning](https://github.com/ChristosChristofidis/awesome-deep-learning#readme) - Neural networks.
 	- [TensorFlow](https://github.com/jtoy/awesome-tensorflow#readme) - Library for machine intelligence.
 	- [TensorFlow.js](https://github.com/aaronhma/awesome-tensorflow-js#readme) - WebGL-accelerated machine learning JavaScript library for training and deploying models.
+	- [TensorFlow Lite](https://github.com/margaretmz/awesome-tflite) - A set of tools to help convert and optimize TensorFlow models for on-device training and inference.
```
```suggestion
	- [TensorFlow Lite](https://github.com/margaretmz/awesome-tflite#readme) - A set of tools to help convert and optimize TensorFlow models for on-device training and inference.
```

And I think the description could be slightly shorter.

margaretmz commented:
@sindresorhus, thanks for your review and detailed feedback! 

I made the following updates accordingly:
* Updated image to 300 dpi and redesigned it to be simpler.
* Linked TensorFlow Lite to https://www.tensorflow.org/lite
* Renamed “TFLite“ -> “TensorFlow Lite“ everywhere.
* Fixed formatting of the “ML Kit examples” section.
* Updated date format.
* Shortened the entry description.

klaufel commented:
Hi! Good list!

Its not necessary added the license name or text to the readme. GitHub already shows the license name at the top of the repo.

margaretmz commented:
> Its not necessary added the license name or text to the readme. GitHub already shows the license name at the top of the repo.

@klaufel - I didn't add the license name to the readme. **Where do you see that?** Any mention of the license names are auto added by GitHub.

margaretmz commented:
@sindresorhus - I made updates based on your feedback on September 28. Let me know if there is anything else I need to do in order to get this PR merged. Thanks!

sindresorhus commented:
You missed https://github.com/sindresorhus/awesome/pull/1865#discussion_r496179725

sindresorhus commented:
The section header icons look weird. They're closer to the previous section.

margaretmz commented:
@sindresorhus thanks for the feedback! I've fixed the missed [#1865 (comment)](https://github.com/sindresorhus/awesome/pull/1865#discussion_r496179725) and removed all section header icons.

Ly0n commented:
The linter shows me just one punctuation error: 
  131:54  List item description must end with proper punctuation  remark-lint:awesome-list-item

The date in the blog posts at the beginning looks a bit strange. Is it really so important that it has to be in the first place? A manual cronological order should be enough with a reference to the reason at the end. Maybe this is also a matter of taste.   

Otherwise an awesome list with many current and good projects.

margaretmz commented:
Thanks @Ly0n for your review!

> The linter shows me just one punctuation error:
> 131:54 List item description must end with proper punctuation remark-lint:awesome-list-item

Great catch! There was a space at the end of line 131 that was causing the awesome-lint error. Fixed it.

> The date in the blog posts at the beginning looks a bit strange. Is it really so important that it has to be in the first place? > .... Maybe this is also a matter of taste.

Yeah sorting chronologically by date is super important for me, from blog posts, to books to research papers etc... With technology in this space moving so fast thing quickly become out-dated.

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1329745274701377542
